src/scripts.js

```diff
@@ -181,15 +180,21 @@ const defineEventListeners = () => {
 
   popupSuccess.addEventListener('click', closePopUp);
 
-  allRecipesTab.onchange = () => {
-    resetTabs(store.recipeRepo.allRecipes);
-  };
+  popupError.addEventListener('click', closePopUp);
 
-  cookbookTab.onchange = () => {
-    resetTabs(store.user.favoriteRecipeRepo.allRecipes);
-  };
+  allRecipesTab.addEventListener('click', (event) => {
+    resetTabs(event, store.recipeRepo.allRecipes);
+  });
+
+  cookbookTab.addEventListener('click', (event) => {
+    resetTabs(event, store.user.favoriteRecipeRepo.allRecipes);
+  });
 };
 
+addIngredientsBtn.addEventListener('click', addToPantry);
+
+cookAfterAddingBtn.addEventListener('click', removeFromPantry);
+
```

Comment:
Created only one instance of event listeners for the `addIngredientsBtn` and `cookAfterAddingBtn`, so we no longer will have the issue with event listeners duplicating and invoking multiple network requests.